pkg/storage/s3/s3.go

```diff
@@ -222,3 +191,56 @@ func generateBucketName(prefix string) string {
 	id := uuid.New().String()
 	return prefix + id
 }
+
+// setInstanceBucketName generates a bucket name for the S3 Bucket, tests to
+// confirm if the bucket is accessible and then updates the instance status
+// with the name
+func setInstanceBucketName(d *driver, s3Client Client, reqLogger logr.Logger, instance *veleroInstallCR.VeleroInstall) error {
+	bucketLog := reqLogger.WithValues("StorageBucket.Name", instance.Status.StorageBucket.Name, "StorageBucket.Region", d.Config.Region)
+
+	// Use an existing bucket, if it exists.
+	bucketLog.Info("No S3 bucket defined. Searching for existing bucket to use")
+	bucketlist, err := ListBucketsInRegion(s3Client, d.Config.Region)
+	if err != nil {
+		return err
+	}
+
+	bucketinfo, err := ListBucketTags(s3Client, bucketlist.Buckets)
+	if err != nil {
+		return err
+	}
+
+	existingBucket := FindMatchingTags(bucketinfo, d.Config.InfraName)
+	if existingBucket != "" {
+		// HEAD the bucket. if it returns ~404 then label it as do-not-recover
```

Comment:
This doesn't work. The problem is that `ListBucketsInRegion` and `ListBucketTags` return errors when the bucket doesn't exist, so we won't even get to this point.

Reply:
Well damn.